src/lib/edge-functions/proxy.js

```diff
@@ -152,34 +146,39 @@ const prepareServer = async ({
   port,
   projectDir,
 }) => {
-  const bundler = await import('@netlify/edge-bundler')
-  const distImportMapPath = getPathInProject([DIST_IMPORT_MAP_PATH])
-  const runIsolate = await bundler.serve({
-    ...getDownloadUpdateFunctions(),
-    certificatePath,
-    debug: env.NETLIFY_DENO_DEBUG === 'true',
-    distImportMapPath,
-    formatExportTypeError: (name) =>
-      `${NETLIFYDEVERR} ${chalk.red('Failed')} to load Edge Function ${chalk.yellow(
-        name,
-      )}. The file does not seem to have a function as the default export.`,
-    formatImportError: (name) => `${NETLIFYDEVERR} ${chalk.red('Failed')} to run Edge Function ${chalk.yellow(name)}:`,
-    importMaps,
-    inspectSettings,
-    port,
-  })
-  const registry = new EdgeFunctionsRegistry({
-    bundler,
-    config,
-    configPath,
-    directories,
-    getUpdatedConfig,
-    internalFunctions,
-    projectDir,
-    runIsolate,
-  })
+  try {
+    const bundler = await import('@netlify/edge-bundler')
+    const distImportMapPath = getPathInProject([DIST_IMPORT_MAP_PATH])
+    const runIsolate = await bundler.serve({
+      ...getDownloadUpdateFunctions(),
+      certificatePath,
+      debug: env.NETLIFY_DENO_DEBUG === 'true',
+      distImportMapPath,
+      formatExportTypeError: (name) =>
+        `${NETLIFYDEVERR} ${chalk.red('Failed')} to load Edge Function ${chalk.yellow(
+          name,
+        )}. The file does not seem to have a function as the default export.`,
+      formatImportError: (name) =>
+        `${NETLIFYDEVERR} ${chalk.red('Failed')} to run Edge Function ${chalk.yellow(name)}:`,
+      importMaps,
+      inspectSettings,
+      port,
+    })
+    const registry = new EdgeFunctionsRegistry({
+      bundler,
+      config,
+      configPath,
+      directories,
+      getUpdatedConfig,
+      internalFunctions,
+      projectDir,
+      runIsolate,
+    })
 
-  return { registry, runIsolate }
+    return { registry, runIsolate }
```

Comment:
Since we're not using `runIsolate` outside of this function, maybe we can `return registry`, which would also simplify things in `initializeProxy` because we could do `const [geoLocation, registry] = await Promise.all(...)` and `await registry.initialize()`?